src/main/java/net/gecore/streamerendpoints/service/twitch/TwitchAPI.java

```diff
@@ -20,18 +17,21 @@
 
   private final RateLimitService ratelimitService;
 
-  private final Logger LOGGER = LoggerFactory.getLogger(TwitchAPI.class);
+//  private final Logger LOGGER = LoggerFactory.getLogger(TwitchAPI.class);
+
+  private final SharedAPI sAPI;
 
   @Autowired
   public TwitchAPI(RateLimitService ratelimitService){
     this.ratelimitService = ratelimitService;
+    sAPI = new SharedAPI();
   }
 
   public String request(URL url, HttpMethod httpMethod, Map<String, String> headers)
       throws TwitchAPIException{
     if(ratelimitService.canPerformRequest()){
-      HttpsURLConnection con = buildConnection(url, httpMethod, headers);
-      return readResponse(con, true);
+      HttpsURLConnection con = sAPI.buildConnection(url, httpMethod, headers);
+      return sAPI.readResponse(con, true);
```

Comment:
And right here  because it is part of the twitchAPI you should get the ApiResponse object returned. 
That object should then call the previously commented out ratelimitservice update call.

For example:
```java
ApiResponse response = sAPI.readResponse(con, true);
ratelimitService.updateRateLimit(response.getHeaders);
```